### PR TITLE
chore: require environments field in project configuration

### PR DESCRIPTION
**Pull request by Orzelius**
Now environment field is required in project configs.
This also eliminates the need for an internal default env (used to be "local"), now if default env is not set just the first one will be selected (as before) and error is thrown if envs is undefined or an empty array.

### CODE REVIEW

File: core/src/cli/cli.ts [hunk 1]
```diff
@@ -66,9 +66,9 @@ import { renderDivider } from "../logger/util"
 import { emoji as nodeEmoji } from "node-emoji"
 
 export async function makeDummyGarden(root: string, gardenOpts: GardenOpts) {
-  const environments = gardenOpts.environmentName
+  const environments: EnvironmentConfig[] = gardenOpts.environmentName
     ? [{ name: parseEnvironment(gardenOpts.environmentName).environment, defaultNamespace, variables: {} }]
-    : defaultEnvironments
+    : [{ defaultNamespace: "default", name: "dummy-env", variables: {} }]
```
Comment (edvald) [inline core/src/cli/cli.ts:71]: I'm not quite sure where but I sort of suspect the `"dummy-env"` string might bleed through somewhere. Maybe just `"default"` instead?

File: core/src/config/project.ts [hunk 2]
```diff
@@ -442,7 +423,8 @@ export function resolveProjectConfig({
       ...globalConfig,
       name,
       defaultEnvironment,
-      environments: [],
+      // environments are validated later
+      environments: [{ defaultNamespace: null, name: "fake-env-only-here-for-inital-load", variables: {} }],
```
Comment (Orzelius) [inline core/src/config/project.ts:427]: I'm not too sure about this line here, if anyone has better ideas I'm all ears

Comment (edvald) [reply core/src/config/project.ts:427]: Hmm. Does this actually need to be changed?

Comment (Orzelius) [reply core/src/config/project.ts:427]: It is, if I leave it to an empty array it won't pass the schema check and if I pass the real ends the vars won't be resolved and I'll get regex match errors

File: core/src/cli/cli.ts (hunk 1 above)
Comment (Orzelius) [reply core/src/cli/cli.ts]: makes sense, fixed in [e0d27aa](https://github.com/garden-io/garden/pull/3078/commits/e0d27aa78210d5e9b4ac17f0caf10333e815be46)

**Review by edvald — APPROVED** [review]

**Comment by edvald** [comment]
Approved 👍  When merging, please tag the squashed commit as `refactor` and include a `BREAKING CHANGE:` clause so that we catch it in changelogs.

**Pull request merged**